src/tools/compiletest/src/runtest.rs

```diff
@@ -1352,67 +1307,89 @@ actual:\n\
             }
         }
 
-        self.compose_and_run(args,
-                             self.props.rustc_env.clone(),
+        rustc.envs(self.props.rustc_env.clone());
+        self.compose_and_run(rustc,
                              self.config.compile_lib_path.to_str().unwrap(),
                              Some(aux_dir.to_str().unwrap()),
-                             input,
-                             None)
+                             input)
     }
 
     fn compose_and_run(&self,
-                       ProcArgs{ args, prog }: ProcArgs,
-                       procenv: Vec<(String, String)> ,
+                       mut command: Command,
                        lib_path: &str,
                        aux_path: Option<&str>,
```

Comment:
I wonder if maybe we could remove these arguments as well? It looks like not all tests need to pass this so we could have a helper which works with these variables and otherwise these aren't passed and the helper isn't called when not needed.